allennlp/training/no_op_trainer.py

```diff
@@ -1,32 +1,76 @@
+import math
 import os
-from typing import Dict, Any
+from typing import Any, Dict, List, Union
 
 from allennlp.common import Params
+from allennlp.common.tqdm import Tqdm
+from allennlp.common.util import lazy_groups_of
 from allennlp.models import Model
+from allennlp.training import util as training_util
 from allennlp.training.checkpointer import Checkpointer
 from allennlp.training.trainer import TrainerPieces
 from allennlp.training.trainer_base import TrainerBase
 
+
 @TrainerBase.register("no_op")
 class NoOpTrainer(TrainerBase):
-    def __init__(self, serialization_dir: str, model: Model) -> None:
+    def __init__(self, serialization_dir: str, model: Model, cuda_device: Union[int, List] = -1) -> None:
         """
         A trivial trainer to assist in making model archives for models that do not actually
         require training. For instance, a majority class baseline.
         """
 
-        super().__init__(serialization_dir, cuda_device=-1)
+        super().__init__(serialization_dir, cuda_device=cuda_device)
         self.model = model
+        # TODO: add iterator and so on.
 
     @classmethod
-    def from_params(cls,   # type: ignore
+    def from_params(cls,  # type: ignore
                     params: Params,
                     serialization_dir: str,
                     recover: bool = False):
         # pylint: disable=arguments-differ
         pieces = TrainerPieces.from_params(params, serialization_dir, recover)  # pylint: disable=no-member
         return NoOpTrainer(serialization_dir, pieces.model)
 
+    def _run_on_validation(self) -> Dict[str, float]:
```

Comment:
This basically copies:

https://github.com/allenai/allennlp/blob/9171766b148fb4e37576c20b2285611354a3a686/allennlp/training/trainer.py#L406-L443